authz/policy/common/src/main/java/org/keycloak/authorization/policy/provider/role/RolePolicyProviderFactory.java

```diff
@@ -210,6 +211,8 @@ private Set<RoleDefinition> getRoles(String rawRoles, RealmModel realm) {
         return Collections.emptySet();
     }
 
+    public static final Pattern UUID_PATTERN = Pattern.compile("[0-9A-Fa-f]{8}-[0-9A-Fa-f]{4}-[0-9A-Fa-f]{4}-[0-9A-Fa-f]{4}-[0-9A-Fa-f]{12}");
+
     private RoleModel getRole(RolePolicyRepresentation.RoleDefinition definition, RealmModel realm) {
         String roleName = definition.getId();
```

Comment:
As an outside contributor, It is weird why `getId` returns `roleName` and we check for both id and name :shrug:

Reply:
It is a mess IMHO. I would hope that in a V2 version of the API we could get rid of it.